source/agora/protocol/GossipProtocol.d

```diff
@@ -0,0 +1,54 @@
+/*******************************************************************************
+
+    Contains the code used for peer-to-peer communication.
+
+    Copyright:
+        Copyright (c) 2019 BOS Platform Foundation Korea
+        All rights reserved.
+
+    License:
+        MIT License. See LICENSE for details.
+
+*******************************************************************************/
+
+module agora.protocol.GossipProtocol;
+import agora.common.Data;
+import agora.protocol.Cache;
+
+alias MessageCache = Cache!(Hash, bool);
+
+/// Interface of Protocol
+interface IProtocol {
```

Comment:
I don't think that's generic enough to need an interface at the moment.

Reply:
Modified.

Reply:
@Geod24 Check Please